File: src/Application Specific/Classes/Entity Collections/Engineer/Engineer.cs
using System;
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Engineers
    {
        public class Engineer
        {
            private DataTypeValidator _dataTypeValidator;

            public Engineer()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            private int _EngineerID = 0;

            public int EngineerID
            {
                get
                {
                    return _EngineerID;
                }
            }

            public object SetEngineerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerID", value);
                }
            }

            private int _RegionID = 0;

            public int RegionID
            {
                get
                {
                    return _RegionID;
                }
            }

            public object SetRegionID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_RegionID", value);
                }
            }

            private string _Name = string.Empty;

            public string Name
            {
                get
                {
                    return _Name;
                }
            }

            public object SetName
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Name", value);
                }
            }

            private string _TelephoneNum = string.Empty;

            public string TelephoneNum
            {
                get
                {
                    return _TelephoneNum;
                }
            }

            public object SetTelephoneNum
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TelephoneNum", value);
                }
            }

            private int _PDAID = 0;

            public int PDAID
            {
                get
                {
                    return _PDAID;
                }
            }

            public object SetPDAID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PDAID", value);
                }
            }

            private bool _Apprentice = false;

            public bool Apprentice
            {
                get
                {
                    return _Apprentice;
                }
            }

            public object SetApprentice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Apprentice", value);
                }
            }

            private double _costToCompanyDouble = 0;

            public double CostToCompanyDouble
            {
                get
                {
                    return _costToCompanyDouble;
                }
            }

            public object SetCostToCompanyDouble
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_costToCompanyDouble", value);
                }
            }

            private double _costToCompanyTimeAndHalf = 0;

            public double CostToCompanyTimeAndHalf
            {
                get
                {
                    return _costToCompanyTimeAndHalf;
                }
            }

            public object SetCostToCompanyTimeAndHalf
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_costToCompanyTimeAndHalf", value);
                }
            }

            private double _costToCompanyNormal = 0;

            public double CostToCompanyNormal
            {
                get
                {
                    return _costToCompanyNormal;
                }
            }

            public object SetCostToCompanyNormal
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_costToCompanyNormal", value);
                }
            }

            private string _nextOfKinName = "";

            public string NextOfKinName
            {
                get
                {
                    return _nextOfKinName;
                }
            }

            public object SetNextOfKinName
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_nextOfKinName", value);
                }
            }

            private string _nextOfKinContact = "";

            public string NextOfKinContact
            {
                get
                {
                    return _nextOfKinContact;
                }
            }

            public object SetNextOfKinContact
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_nextOfKinContact", value);
                }
            }

            private string _startingDetails = "";

            public string StartingDetails
            {
                get
                {
                    return _startingDetails;
                }
            }

            public object SetStartingDetails
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_startingDetails", value);
                }
            }

            private string _drivingLicenceNo = "";

            public string DrivingLicenceNo
            {
                get
                {
                    return _drivingLicenceNo;
                }
            }

            public object SetDrivingLicenceNo
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_drivingLicenceNo", value);
                }
            }

            private DateTime _drivingLicenceIssueDate = default;

            public DateTime DrivingLicenceIssueDate
            {
                get
                {
                    return _drivingLicenceIssueDate;
                }

                set
                {
                    _drivingLicenceIssueDate = value;
                }
            }

            private int _payGradeID = 0;

            public int PayGradeID
            {
                get
                {
                    return _payGradeID;
                }
            }

            public object SetPayGradeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_payGradeID", value);
                }
            }

            private decimal _annualSalary = 0;

            public decimal AnnualSalary
            {
                get
                {
                    return _annualSalary;
                }
            }

            public object SetAnnualSalary
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_annualSalary", value);
                }
            }

            private string _nINumber = "";

            public string NINumber
            {
                get
                {
                    return _nINumber;
                }
            }

            public object SetNINumber
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_nINumber", value);
                }
            }

            private string _holidayYearStart = "01/01";

            public string HolidayYearStart
            {
                get
                {
                    return _holidayYearStart;
                }
            }

            public object SetHolidayYearStart
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_holidayYearStart", value);
                }
            }

            private string _holidayYearEnd = "31/12";

            public string HolidayYearEnd
            {
                get
                {
                    return _holidayYearEnd;
                }
            }

            public object SetHolidayYearEnd
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_holidayYearEnd", value);
                }
            }

            private decimal _daysHolidayAllowed = 0;

            public decimal DaysHolidayAllowed
            {
                get
                {
                    return _daysHolidayAllowed;
                }
            }

            public object SetDaysHolidayAllowed
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_daysHolidayAllowed", value);
                }
            }

            private string _gasSafeNo = "";

            public string GasSafeNo
            {
                get
                {
                    return _gasSafeNo;
                }
            }

            public object SetGasSafeNo
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_gasSafeNo", value);
                }
            }

            private string _oftecNo = "";

            public string OftecNo
            {
                get
                {
                    return _oftecNo;
                }
            }

            public object SetOftecNo
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_oftecNo", value);
                }
            }

            private int _engineerGroupID = 0;

            public int EngineerGroupID
            {
                get
                {
                    return _engineerGroupID;
                }
            }

            public object SetEngineerGroupID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_engineerGroupID", value);
                }
            }

            private string _supervisor = "";

            public string Supervisor
            {
                get
                {
                    return _supervisor;
                }
            }

            public object SetSupervisor
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_supervisor", value);
                }
            }

            private string _technician = "";

            public string Technician
            {
                get
                {
                    return _technician;
                }
            }

            public object SetTechnician
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_technician", value);
                }
            }

            private string _Department = "";

            public string Department
            {
                get
                {
                    return _Department;
                }
            }

            public object SetDepartment
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Department", value);
                }
            }

            private int _ServPri = 0;

            public int ServPri
            {
                get
                {
                    return _ServPri;
                }
            }

            public object SetServPri
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ServPri", value);
                }
            }

            private int _BreakPri = 0;

            public int BreakPri
            {
                get
                {
                    return _BreakPri;
                }
            }

            public object SetBreakPri
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_BreakPri", value);
                }
            }

            private int _DailyServiceLimit = 0;

            public int DailyServiceLimit
            {
                get
                {
                    return _DailyServiceLimit;
                }
            }

            public object SetDailyServiceLimit
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_DailyServiceLimit", value);
                }
            }

            private string _HomePostcode = "";

            public string HomePostcode
            {
                get
                {
                    return _HomePostcode;
                }
            }

            public object SetHomePostcode
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_HomePostcode", value);
                }
            }

            private decimal _Longitude = 0;

            public decimal Longitude
            {
                get
                {
                    return _Longitude;
                }
            }

            public object SetLongitude
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Longitude", value);
                }
            }

            private decimal _Latitude = 0;

            public decimal Latitude
            {
                get
                {
                    return _Latitude;
                }
            }

            public object SetLatitude
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Latitude", value);
                }
            }

            private int _ManagerUserID = Conversions.ToInteger(false);

            public int ManagerUserID
            {
                get
                {
                    return _ManagerUserID;
                }
            }

            public object SetManagerUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ManagerUserID", value);
                }
            }

            private string _WebAppPassword = "";

            public string WebAppPassword
            {
                get
                {
                    return _WebAppPassword;
                }
            }

            public object SetWebAppPassword
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_WebAppPassword", value);
                }
            }

            private int _ragRating = 0;

            public int RagRating
            {
                get
                {
                    return _ragRating;
                }
            }

            public object SetRagRating
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ragRating", value);
                }
            }

            private int _UserID = 0;

            public int UserID
            {
                get
                {
                    return _UserID;
                }
            }

            public object SetUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_UserID", value);
                }
            }

            public DateTime RagDate { get; set; } = default;

            private decimal _visitSpendLimit = default;

            public decimal VisitSpendLimit
            {
                get
                {
                    return _visitSpendLimit;
                }
            }

            public object SetVisitSpendLimit
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_visitSpendLimit", value);
                }
            }

            private decimal _engineerRoleId = default;

            public int EngineerRoleId
            {
                get
                {
                    return Conversions.ToInteger(_engineerRoleId);
                }
            }

            public object SetEngineerRoleId
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_engineerRoleId", value);
                }
            }

            private string _emailAddress = string.Empty;

            public string EmailAddress
            {
                get
                {
                    return _emailAddress;
                }
            }

            public object SetEmailAddress
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_emailAddress", value);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Engineer/Engineer.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace Engineers

        Public Class Engineer

            Private _dataTypeValidator As DataTypeValidator

            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False

            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False

            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property

            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Engineer Properties"

            Private _EngineerID As Integer = 0

            Public ReadOnly Property EngineerID() As Integer
                Get
                    Return _EngineerID
                End Get
            End Property

            Public WriteOnly Property SetEngineerID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_EngineerID", Value)
                End Set
            End Property

            Private _RegionID As Integer = 0

            Public ReadOnly Property RegionID() As Integer
                Get
                    Return _RegionID
                End Get
            End Property

            Public WriteOnly Property SetRegionID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_RegionID", Value)
                End Set
            End Property

            Private _Name As String = String.Empty

            Public ReadOnly Property Name() As String
                Get
                    Return _Name
                End Get
            End Property

            Public WriteOnly Property SetName() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Name", Value)
                End Set
            End Property

            Private _TelephoneNum As String = String.Empty

            Public ReadOnly Property TelephoneNum() As String
                Get
                    Return _TelephoneNum
                End Get
            End Property

            Public WriteOnly Property SetTelephoneNum() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TelephoneNum", Value)
                End Set
            End Property

            Private _PDAID As Integer = 0

            Public ReadOnly Property PDAID() As Integer
                Get
                    Return _PDAID
                End Get
            End Property

            Public WriteOnly Property SetPDAID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PDAID", Value)
                End Set
            End Property

            Private _Apprentice As Boolean = False

            Public ReadOnly Property Apprentice() As Boolean
                Get
                    Return _Apprentice
                End Get
            End Property

            Public WriteOnly Property SetApprentice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Apprentice", Value)
                End Set
            End Property

            Private _costToCompanyDouble As Double = 0

            Public ReadOnly Property CostToCompanyDouble() As Double
                Get
                    Return _costToCompanyDouble
                End Get
            End Property

            Public WriteOnly Property SetCostToCompanyDouble() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_costToCompanyDouble", Value)
                End Set
            End Property

            Private _costToCompanyTimeAndHalf As Double = 0

            Public ReadOnly Property CostToCompanyTimeAndHalf() As Double
                Get
                    Return _costToCompanyTimeAndHalf
                End Get
            End Property

            Public WriteOnly Property SetCostToCompanyTimeAndHalf() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_costToCompanyTimeAndHalf", Value)
                End Set
            End Property

            Private _costToCompanyNormal As Double = 0

            Public ReadOnly Property CostToCompanyNormal() As Double
                Get
                    Return _costToCompanyNormal
                End Get
            End Property

            Public WriteOnly Property SetCostToCompanyNormal() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_costToCompanyNormal", Value)
                End Set
            End Property

            Private _nextOfKinName As String = ""

            Public ReadOnly Property NextOfKinName() As String
                Get
                    Return _nextOfKinName
                End Get
            End Property

            Public WriteOnly Property SetNextOfKinName() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_nextOfKinName", Value)
                End Set
            End Property

            Private _nextOfKinContact As String = ""

            Public ReadOnly Property NextOfKinContact() As String
                Get
                    Return _nextOfKinContact
                End Get
            End Property

            Public WriteOnly Property SetNextOfKinContact() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_nextOfKinContact", Value)
                End Set
            End Property

            Private _startingDetails As String = ""

            Public ReadOnly Property StartingDetails() As String
                Get
                    Return _startingDetails
                End Get
            End Property

            Public WriteOnly Property SetStartingDetails() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_startingDetails", Value)
                End Set
            End Property

            Private _drivingLicenceNo As String = ""

            Public ReadOnly Property DrivingLicenceNo() As String
                Get
                    Return _drivingLicenceNo
                End Get
            End Property

            Public WriteOnly Property SetDrivingLicenceNo() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_drivingLicenceNo", Value)
                End Set
            End Property

            Private _drivingLicenceIssueDate As DateTime = Nothing

            Public Property DrivingLicenceIssueDate() As DateTime
                Get
                    Return _drivingLicenceIssueDate
                End Get
                Set(ByVal Value As DateTime)
                    _drivingLicenceIssueDate = Value
                End Set
            End Property

            Private _payGradeID As Integer = 0

            Public ReadOnly Property PayGradeID() As Integer
                Get
                    Return _payGradeID
                End Get
            End Property

            Public WriteOnly Property SetPayGradeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_payGradeID", Value)
                End Set
            End Property

            Private _annualSalary As Decimal = 0

            Public ReadOnly Property AnnualSalary() As Decimal
                Get
                    Return _annualSalary
                End Get
            End Property

            Public WriteOnly Property SetAnnualSalary() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_annualSalary", Value)
                End Set
            End Property

            Private _nINumber As String = ""

            Public ReadOnly Property NINumber() As String
                Get
                    Return _nINumber
                End Get
            End Property

            Public WriteOnly Property SetNINumber() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_nINumber", Value)
                End Set
            End Property

            Private _holidayYearStart As String = "01/01"

            Public ReadOnly Property HolidayYearStart() As String
                Get
                    Return _holidayYearStart
                End Get
            End Property

            Public WriteOnly Property SetHolidayYearStart() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_holidayYearStart", Value)
                End Set
            End Property

            Private _holidayYearEnd As String = "31/12"

            Public ReadOnly Property HolidayYearEnd() As String
                Get
                    Return _holidayYearEnd
                End Get
            End Property

            Public WriteOnly Property SetHolidayYearEnd() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_holidayYearEnd", Value)
                End Set
            End Property

            Private _daysHolidayAllowed As Decimal = 0

            Public ReadOnly Property DaysHolidayAllowed() As Decimal
                Get
                    Return _daysHolidayAllowed
                End Get
            End Property

            Public WriteOnly Property SetDaysHolidayAllowed() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_daysHolidayAllowed", Value)
                End Set
            End Property

            Private _gasSafeNo As String = ""

            Public ReadOnly Property GasSafeNo() As String
                Get
                    Return _gasSafeNo
                End Get
            End Property

            Public WriteOnly Property SetGasSafeNo() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_gasSafeNo", Value)
                End Set
            End Property

            Private _oftecNo As String = ""

            Public ReadOnly Property OftecNo() As String
                Get
                    Return _oftecNo
                End Get
            End Property

            Public WriteOnly Property SetOftecNo() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_oftecNo", Value)
                End Set
            End Property

            Private _engineerGroupID As Integer = 0

            Public ReadOnly Property EngineerGroupID() As Integer
                Get
                    Return _engineerGroupID
                End Get
            End Property

            Public WriteOnly Property SetEngineerGroupID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_engineerGroupID", Value)
                End Set
            End Property

            Private _supervisor As String = ""

            Public ReadOnly Property Supervisor() As String
                Get
                    Return _supervisor
                End Get
            End Property

            Public WriteOnly Property SetSupervisor() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_supervisor", Value)
                End Set
            End Property

            Private _technician As String = ""

            Public ReadOnly Property Technician() As String
                Get
                    Return _technician
                End Get
            End Property

            Public WriteOnly Property SetTechnician() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_technician", Value)
                End Set
            End Property

            Private _Department As String = ""

            Public ReadOnly Property Department() As String
                Get
                    Return _Department
                End Get
            End Property

            Public WriteOnly Property SetDepartment() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Department", Value)
                End Set
            End Property

            Private _ServPri As Integer = 0

            Public ReadOnly Property ServPri() As Integer
                Get
                    Return _ServPri
                End Get
            End Property

            Public WriteOnly Property SetServPri() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ServPri", Value)
                End Set
            End Property

            Private _BreakPri As Integer = 0

            Public ReadOnly Property BreakPri() As Integer
                Get
                    Return _BreakPri
                End Get
            End Property

            Public WriteOnly Property SetBreakPri() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_BreakPri", Value)
                End Set
            End Property

            Private _DailyServiceLimit As Integer = 0

            Public ReadOnly Property DailyServiceLimit() As Integer
                Get
                    Return _DailyServiceLimit
                End Get
            End Property

            Public WriteOnly Property SetDailyServiceLimit() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_DailyServiceLimit", Value)
                End Set
            End Property

            Private _HomePostcode As String = ""

            Public ReadOnly Property HomePostcode() As String
                Get
                    Return _HomePostcode
                End Get
            End Property

            Public WriteOnly Property SetHomePostcode() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_HomePostcode", Value)
                End Set
            End Property

            Private _Longitude As Decimal = 0

            Public ReadOnly Property Longitude() As Decimal
                Get
                    Return _Longitude
                End Get
            End Property

            Public WriteOnly Property SetLongitude() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Longitude", Value)
                End Set
            End Property

            Private _Latitude As Decimal = 0

            Public ReadOnly Property Latitude() As Decimal
                Get
                    Return _Latitude
                End Get
            End Property

            Public WriteOnly Property SetLatitude() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Latitude", Value)
                End Set
            End Property

            Private _ManagerUserID As Integer = False

            Public ReadOnly Property ManagerUserID() As Integer
                Get
                    Return _ManagerUserID
                End Get
            End Property

            Public WriteOnly Property SetManagerUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ManagerUserID", Value)
                End Set
            End Property

            Private _WebAppPassword As String = ""

            Public ReadOnly Property WebAppPassword() As String
                Get
                    Return _WebAppPassword
                End Get
            End Property

            Public WriteOnly Property SetWebAppPassword() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_WebAppPassword", Value)
                End Set
            End Property

            Private _ragRating As Integer = 0

            Public ReadOnly Property RagRating() As Integer
                Get
                    Return _ragRating
                End Get
            End Property

            Public WriteOnly Property SetRagRating() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ragRating", Value)
                End Set
            End Property

            Private _UserID As Integer = 0

            Public ReadOnly Property UserID() As Integer
                Get
                    Return _UserID
                End Get
            End Property

            Public WriteOnly Property SetUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_UserID", Value)
                End Set
            End Property

            Public Property RagDate As DateTime = Nothing

            Private _visitSpendLimit As Decimal = Nothing

            Public ReadOnly Property VisitSpendLimit() As Decimal
                Get
                    Return _visitSpendLimit
                End Get
            End Property

            Public WriteOnly Property SetVisitSpendLimit() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_visitSpendLimit", Value)
                End Set
            End Property

            Private _engineerRoleId As Decimal = Nothing

            Public ReadOnly Property EngineerRoleId() As Integer
                Get
                    Return _engineerRoleId
                End Get
            End Property

            Public WriteOnly Property SetEngineerRoleId() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_engineerRoleId", Value)
                End Set
            End Property

            Private _emailAddress As String = String.Empty

            Public ReadOnly Property EmailAddress() As String
                Get
                    Return _emailAddress
                End Get
            End Property

            Public WriteOnly Property SetEmailAddress() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_emailAddress", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace